File: eCar/eCar.Services/Database/DriverVehicle.cs
using System;
using System.Collections.Generic;

namespace eCar.Services.Database;

public partial class DriverVehicle
{
    public int Id { get; set; }

    public int DriverId { get; set; }

    public int VehicleId { get; set; }

    public DateTime? DatePickUp { get; set; }
    public DateTime? DateDropOff { get; set; }

    public virtual Driver Driver { get; set; } = null!;

    public virtual Vehicle Vehicle { get; set; } = null!;
}

File: eCar/UI/ecar_admin/lib/models/DriverVehicle/driverVehicle.dart
import 'package:ecar_admin/models/Driver/driver.dart';
import 'package:ecar_admin/models/Vehicle/vehicle.dart';
import 'package:json_annotation/json_annotation.dart';

part 'driverVehicle.g.dart';

@JsonSerializable()
class DriverVehicle {
  int? id;
  int? driverId;
  int? vehicleId;
  DateTime? datePickUp;
  DateTime? dateDropOff;
  Driver? driver;
  Vehicle? vehicle;

  DriverVehicle(
      {this.id,
      this.driverId,
      this.vehicleId,
      this.datePickUp,
      this.dateDropOff,
      this.driver,
      this.vehicle});

  factory DriverVehicle.fromJson(Map<String, dynamic> json) =>
      _$DriverVehicleFromJson(json);

  Map<String, dynamic> toJson() => _$DriverVehicleToJson(this);
}
